File: csharp/Chapter06/Person.cs
using System;

namespace com.db4o.dg2db4o.chapter6
{
    public class Person : IComparable
    {
        private string _name;
        private int _age;

        public Person() { }

        public Person(String name, int age)
        {
            _name = name;
            _age = age;
        }


        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public int Age
        {
            get
            {
                return _age;
            }
            set
            {
                _age = value;
            }
        }

        public override string ToString()
        {
            return "[" + Name + ";" + Age + "]";
        }

        public int CompareTo(Object o)
        {
            if (o is Person)
            {
                Person p = (Person)o;
                return this.Name.CompareTo(p.Name);
            }
            throw new ArgumentException("object is not a Person");
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter10/Person.java
package com.db4o.dg2db4o.chapter10;

public class Person 
{
    private String _name;
    private int _age;

    public Person(){}
    
    public Person(String name, int age) 
    {
        _name = name;
        _age = age;
    }

    public int getAge() 
    {
       return _age;
    }

    public void setAge(int value) 
    {
        _age = value;
    }

    public String getName() 
    {
        return _name;
    }

    public void setName(String value) 
    {
        _name = value;
    }

    public String toString() 
    {
        return "[" + _name + ";"+ _age + "]";
    }
}
